test/lib/autosetup_helpers_test.rb

```diff
@@ -303,6 +297,50 @@ class DummyClient < Yast::Client
     end
   end
 
+  describe "#method_missing" do
+    let(:profile) { general_section }
+    let(:general_section) { { "general" => { "semi-automatic" => ["networking"] } } }
+    let(:register_section) { { "suse_register" => { "reg_code" => "12345" } } }
+
+    before do
+      Yast::Profile.current = profile
+    end
+
+    context "when the method name matchs /(?<section_name>.*)_section/'" do
+      it "returns the profile section with the matched :section_name when present" do
+        expect(client.general_section).to eql(general_section["general"])
+      end
+
+      it "returns an empty hash when the section with the matched :section_name is not present" do
+        expect(client.networking_section).to eql({})
+      end
+    end
+
+    context "when the method name matchs /(?<section_name>.*)_section?/'" do
+      it "returns true if the profile section with the matched name is defined" do
+        expect(client.general_section?).to eql(true)
+      end
+
+      it "returns false if the matched section name is not defined in the profile" do
+        expect(client.suse_register_section?).to eql(false)
+      end
+    end
+  end
+
+  describe "#respond_to_missing?" do
+    it "returns true if the method name matchs /(?<section_name>.*)_section/'" do
+      expect(client.respond_to?("general_section")).to eql(true)
+    end
+
+    it "returns true if the method name matchs /(?<section_name>.*)_section?/'" do
+      expect(client.respond_to?("suse_register_section")).to eql(true)
+    end
+
+    it "returns false otherwise" do
+      expect(client.respond_to?("wrong_section_method_name")).to eql(false)
+    end
+  end
+
```

Comment:
The two tests below should or could be removed if we confirm that the method_missing approach is ok.